mlflow/utils/autologging_utils.py

```diff
@@ -306,6 +310,97 @@ def batch_metrics_logger(run_id):
     batch_metrics_logger.flush()
 
 
+def _check_version_in_range(ver, min_ver, max_ver):
+    return LooseVersion(min_ver) <= LooseVersion(ver) <= LooseVersion(max_ver)
+
+
+def _load_version_file_as_dict():
+    version_file_path = resource_filename(__name__, "../ml-package-versions.yml")
+    with open(version_file_path) as f:
+        return yaml.load(f, Loader=yaml.SafeLoader)
+
+
+_module_version_info_dict = _load_version_file_as_dict()
+
+
+# A map FLAVOR_NAME -> list of dependent (module_name, key_in_version_file_json)
+_cross_tested_flavor_to_module_name_and_module_key = {
+    "fastai": [("fastai", "fastai-1.x")],
+    "gluon": [("mxnet", "gluon")],
+    "keras": [("keras", "keras")],
+    "lightgbm": [("lightgbm", "lightgbm")],
+    "statsmodels": [("statsmodels", "statsmodels")],
+    "tensorflow": [("tensorflow", "tensorflow")],
+    "xgboost": [("xgboost", "xgboost")],
+    "sklearn": [("sklearn", "sklearn")],
+    "pytorch": [("torch", "pytorch"), ("pytorch_lightning", "pytorch-lightning")],
```

Comment:
@WeichenXu123 @harupy I think it's sufficient for us to only map `pytorch` to `pytorch-lightning`, since `torch` is a requirement for `pytorch-lightning` and, therefore, installing `pytorch-lightning` will pull in a compatible version of `torch`.

Accordingly, I think we can make this a map from string to tuple, rather than a map from string to a list of tuples. I also think we can drop the `autologging` subsection from the `pytorch` section of `ml-package-versions.yaml`: https://github.com/mlflow/mlflow/blob/ab6fa411a4237c01e19b3b9747179c85f6fd137e/ml-package-versions.yml#L35-L40

Reply:
@dbczumar Thanks for pointing this out. Agree with dropping the autologging from `pytorch`. Let me just open a PR to address this.